### PR TITLE
Siddhi editor design-view's form changes

### PR DESCRIPTION
**Pull request by PradeepaJey1**

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TriggerConfigGenerator.java [hunk 1]
```diff
@@ -48,11 +49,14 @@ public TriggerConfigGenerator(String siddhiAppString, Map<String, StreamDefiniti
      */
     public TriggerConfig generateTriggerConfig(TriggerDefinition triggerDefinition) throws DesignGenerationException {
         String at = "";
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TriggerConfigGenerator.java]: shall we change the parameter name "at" for a descriptive name please

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TableConfigGenerator.java [hunk 2]
```diff
@@ -55,7 +57,8 @@ public TableConfig generateTableConfig(TableDefinition tableDefinition) throws D
                 tableDefinition.getId(),
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TableConfigGenerator.java]: unnecessary new lines. Please format

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/index.html [hunk 3]
```diff
@@ -1692,52 +1777,71 @@ <h4 class="modal-title file-dialog-title" id="sampleEvent">Generate sample event
...
-                }
+            {
+                predefined: true,
+                name: 'Async',
+                class: 'optional-annotation',
+                isMandatory: false,
+                elements: [
+                    {
+                        key: 'buffer.size',
+                        value: '256',
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/index.html]: lets have it as defaultValue

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/JSON-validator.js [hunk 4]
```diff
@@ -209,15 +209,16 @@ define(['require', 'log', 'jquery', 'lodash', 'designViewUtils'],
...
          * @param element element json
-         * @param type stream or table
+         * @param type stream or table or window or trigger or function
          * @param doNotShowErrorMessages If true error messages will not be shown as alerts. Only the validity will be
          * returned
          * @returns {boolean} validity of the json
          */
-        JSONValidator.prototype.validateStreamOrTable = function (element, type, doNotShowErrorMessages) {
+        JSONValidator.prototype.validateForElementName = function (element, type,
+            doNotShowErrorMessages) {
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/JSON-validator.js]: why new line ?

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/drop-elements.js [hunk 5]
```diff
@@ -724,7 +731,16 @@ define(['require', 'log', 'lodash', 'jquery', 'partition', 'stream', 'query', 'f
             if (isCodeToDesignMode) {
                 name = functionName;
             } else {
-                name = self.formBuilder.DefineFunction(i);
+                //add the new function element to function list
+				var functionOptions = {};
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/drop-elements.js]: format your code

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TriggerConfigGenerator.java (hunk 1 above)
Comment (PradeepaJey1) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TriggerConfigGenerator.java]: Fixed in eceb77b51dbb904e257a9b9b5f72d8e14560d910

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TableConfigGenerator.java (hunk 2 above)
Comment (PradeepaJey1) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/designgenerator/generators/TableConfigGenerator.java]: Fixed in 97aff2523a4468759c33ce94e34dd1c88ea092af

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/JSON-validator.js (hunk 4 above)
Comment (PradeepaJey1) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/JSON-validator.js]: Fixed in 97aff2523a4468759c33ce94e34dd1c88ea092af

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/drop-elements.js (hunk 5 above)
Comment (PradeepaJey1) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/drop-elements.js]: Fixed in 97aff2523a4468759c33ce94e34dd1c88ea092af

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/index.html (hunk 3 above)
Comment (PradeepaJey1) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/index.html]: Fixed in a046ddb3c214ec608ebdf90386019c42b98c4ddc

**Pull request merged**